sktime/classification/base.py

```diff
@@ -131,7 +134,213 @@ def __rmul__(self, other):
             return NotImplemented
 
     def fit(self, X, y):
-        """Fit time series classifier to training data.
+        """
+        Fit time series classifier to training data.
+
+        Parameters
+        ----------
+        X : 3D np.array (any number of dimensions, equal length series)
+                of shape [n_instances, n_dimensions, series_length]
+            or 2D np.array (univariate, equal length series)
+                of shape [n_instances, series_length]
+            or pd.DataFrame with each column a dimension, each cell a pd.Series
+                (any number of dimensions, equal or unequal length series)
+            or of any other supported Panel mtype
+                for list of mtypes, see datatypes.SCITYPE_REGISTER
+                for specifications, see examples/AA_datatypes_and_datasets.ipynb
+        y : 1D np.array of int, of shape [n_instances] - class labels for fitting
+            indices correspond to instance indices in X
+
+        Returns
+        -------
+        estimators : list of reference to a single instance or estimator instances.
+        """
+        self._y = y
+        self._X = X
+        metadata = self._check_classifier_input(X, y)
+        if self.get_tag("capability:multiouput"):
+            self._fit_instance(X, y, metadata)
+            self.estimators = [self]
+            return self
+        else:
+            vdf = VectorizedDF(
+                X=[X] * len(y.columns),
+                y=[(lambda col: y[col])(col) for col in y.columns],
+            )
+            self.estimators = vdf.vectorize_est(estimator=self, method="clone")
+            for col, est in zip(y.columns, self.estimators):
```

Comment:
should this not be replaced by `vectorize_est`, `method="fit"`?

Reply:
I thought of two ways of approaching this issue:
1. Create instances of unfitted estimator using `method='clone'`, fit each of them individually, predict individually.
2. Use `vectorize_est` with `method='fit'` and `method='predict'` across the columns of y.

I am not sure which one is better?

Reply:
2. is the programmatic way to do 1. in sktime, the individual fits/predicts are stored in the returned data frame, and `vectorize_est` also gives access to arbitrary parallel backends